src/Ztm.WebApi/Controllers/ControllerHelper.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Ztm.WebApi.Callbacks;
+
+namespace Ztm.WebApi.Controllers
+{
+    public class ControllerHelper
+    {
+        readonly ICallbackRepository callbackRepository;
+
+        public ControllerHelper(
+            ICallbackRepository callbackRepository)
+        {
+            if (callbackRepository == null)
+            {
+                throw new ArgumentNullException(nameof(callbackRepository));
+            }
+
+            this.callbackRepository = callbackRepository;
+        }
+
+        public async Task<Callback> TryAddCallbackAsync(ControllerBase controller, CancellationToken cancellationToken)
```

Comment:
The name of method is quite confusing. I think `RegisterCallbackAsync` looks better.